api/server/router/build/build_routes.go

```diff
@@ -61,9 +61,21 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 		BuildID:        r.FormValue("buildid"),
 	}
 
-	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
-		// SecurityOpt only supports "credentials-spec" on Windows, and not used on other platforms.
-		return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+	if len(options.SecurityOpt) > 0 {
+		// SecurityOpt only supports "credentialspec" on Windows, and not used on other platforms.
+		if runtime.GOOS != "windows" {
+			return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+		}
+		for _, secOpt := range options.SecurityOpt {
+			opt, _, ok := strings.Cut(secOpt, "=")
+			if !ok {
+				return nil, invalidParam{errors.New("invalid security option: no equals sign in supplied value " + secOpt)}
```

Comment:
```suggestion
				return nil, invalidParam{fmt.Errorf("invalid security option: no equals sign in supplied value %q", secOpt)}
```
You never know what weird strings a user might feed into the API.